api/models/app.go

```diff
@@ -520,30 +537,53 @@ func (a *App) RunAttached(process, command, releaseId string, height, width int,
 					}
 				}
 
-				if *container.Name == process && releaseMatch {
+				if *releaseContainer.Name == process && releaseMatch {
 					break ContainerCheck
 				}
-				container = nil
+				releaseContainer = nil
 			}
+			////////////////////////////////////////////////////
 
-			if container != nil {
+			if releaseContainer != nil {
+				container = releaseContainer
 				break
 			}
 		}
-	}
 
-	if container == nil {
-		return fmt.Errorf("unable to find contaier for %s", process)
+		// If container is nil, the release most likely hasn't been promoted and thus no TaskDefinition for it.
+		if releaseContainer == nil {
+			fmt.Printf("Unable to find contaier for %s. Basing container off of most recent release: %s.\n", process, a.Release)
+			unpromotedRelease = true
+		}
 	}
 
 	ea := make([]string, 0)
 	for _, env := range container.Environment {
 		ea = append(ea, fmt.Sprintf("%s=%s", *env.Name, *env.Value))
 	}
 
-	release, err := GetRelease(a.Name, releaseId)
-	if err != nil {
-		return err
+	// Update any environment variables that might be part of the unpromoted release.
+	if unpromotedRelease {
+		releaseEnv := strings.Split(release.Env, "\n")
+
+	ReleaseEnvCheck:
```

Comment:
That's the gist of the logic, yes. I'll take another look at it to see what else can be improved. It was probably a little hack keeping the environment variables as strings instead of converting them to real maps to make it easier to work with.